HealthcareSystem/GraphicalEditor/MainWindow.xaml.cs

```diff
@@ -239,33 +241,93 @@ private void Canvas_MouseDown(object sender, MouseButtonEventArgs e)
             ChangeEditButtonVisibility();
         }
 
-        private void ApplyShadowEffectToObject(MapObject selectedMapObject)
+        private void ApplyHoverEffectToObject(MapObject hoverMapObject)
         {
-            if (selectedMapObject != null)
+            if (hoverMapObject != null)
             {
-                if (selectedMapObject.MapObjectEntity.MapObjectType.TypeOfMapObject != TypeOfMapObject.ROAD) 
+                if (hoverMapObject.MapObjectEntity.MapObjectType.TypeOfMapObject != TypeOfMapObject.ROAD) 
                 {
-                    selectedMapObject.Rectangle.Effect = new DropShadowEffect { Direction = 0, ShadowDepth = 0, BlurRadius = 14, Opacity = 1, Color = Colors.MediumPurple }; 
+                    ApplyShadowToObjectAndIncreaseZIndex(hoverMapObject);
                 }
               
+                foreach (MapObject mapObject in _allMapObjects)
+                {
+                    if (!mapObject.Equals(hoverMapObject) && !mapObject.Equals(_selectedMapObject))
+                    {
+                        RemoveShadowFromObjectAndDecreaseZIndex(mapObject);
+                    }
+                }
+            }
+            else
+            {
+                foreach (MapObject mapObject in _allMapObjects)
+                {
+                    if (!mapObject.Equals(_selectedMapObject))
+                    {
+                        RemoveShadowFromObjectAndDecreaseZIndex(mapObject);
+                    }
+                }
+            }
+        }
+
+
+        private void ApplySelectionEffectToObject(MapObject selectedMapObject)
```

Comment:
One simple way to make this cleaner (as well as ApplyHoverEffectToObject) would be to first remove the effect from all objects (since you're doing that anyway, whether or not  the object you're applying to is null), and then check if the argument is null, and if it isn't, apply the effect to it. I don't insist on making this change right now though, I'm not sure if that one moment when no object has the effect applied will make an odd visible twitch when the change is happening.

Reply:
Good point. I think that I have successfully implemented your advice in my new commit. Check it out. 😄